data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java

```diff
@@ -0,0 +1,197 @@
+
+        assertThat(expression, isExpression(isParseTree(
+                CONDITIONAL_EXPRESSION,
+                EQUALITY_OPERATOR_EXPRESSION
+        ).withChildrenMatching(
+                isUnaryTree(),
+                isOperator(EQUALITY_OPERATOR),
+                isUnaryTree()
+        )));
+        assertThat(errorListener.isErrorFound(), is(false));
+
+    }
+
+    @Test
+    void testJsonPointerEscapeCharacters() {
+        final ParserRuleContext expression = parseExpression("\"/a b/\\\"c~d\\\"/\\/\"");
+        assertThat(expression, isExpression(isJsonPointerUnaryTree()));
+        assertThat(errorListener.isErrorFound(), is(false));
+    }
+
+    @Test
+    void testRelationalExpression() {
+        final ParserRuleContext expression = parseExpression("1 < 2");
+
+        assertThat(expression, isExpression(isParseTree(
+                CONDITIONAL_EXPRESSION,
+                EQUALITY_OPERATOR_EXPRESSION,
+                REGEX_OPERATOR_EXPRESSION,
+                RELATIONAL_OPERATOR_EXPRESSION
+        ).withChildrenMatching(
+                isUnaryTree(),
+                isOperator(RELATIONAL_OPERATOR),
+                isUnaryTree()
+        )));
+        assertThat(errorListener.isErrorFound(), is(false));
+    }
+
+    @Test
+    void testNestedConditionalExpression() {
+        final ParserRuleContext expression = parseExpression("1 < 2 or 3 <= 4 or 5 > 6 or 7 >= 8");
+
+        final DiagnosingMatcher<ParseTree> relationalOperatorExpression = isParseTree(
+                EQUALITY_OPERATOR_EXPRESSION,
+                REGEX_OPERATOR_EXPRESSION,
+                RELATIONAL_OPERATOR_EXPRESSION
+        ).withChildrenMatching(
+                isUnaryTree(),
+                isOperator(RELATIONAL_OPERATOR),
+                isUnaryTree()
+        );
+        final DiagnosingMatcher<ParseTree> firstConditional = hasContext(
+                CONDITIONAL_EXPRESSION,
+                hasContext(CONDITIONAL_EXPRESSION, relationalOperatorExpression),
+                isOperator(CONDITIONAL_OPERATOR),
+                relationalOperatorExpression
+        );
+        final DiagnosingMatcher<ParseTree> secondConditional = hasContext(
+                CONDITIONAL_EXPRESSION,
+                firstConditional,
+                isOperator(CONDITIONAL_OPERATOR),
+                relationalOperatorExpression
+        );
+        final DiagnosingMatcher<ParseTree> thirdConditional = hasContext(
+                CONDITIONAL_EXPRESSION,
+                secondConditional,
+                isOperator(CONDITIONAL_OPERATOR),
+                relationalOperatorExpression
+        );
+
+        assertThat(expression, isExpression(thirdConditional));
+        assertThat(errorListener.isErrorFound(), is(false));
+    }
+
+    @Test
+    void testValidJsonPointerCharacterSet() {
+        final ParserRuleContext expression = parseExpression(
+                "/ABCDEFGHIJKLMNOPQRSTUVWXYZ/ambcdefghijklmnopqrstuvwxyz/0123456789/_");
```

Comment:
looks like you have an errant `m` after `a`

Reply:
Good catch!